modules/configuration/src/test/java/org/apache/ignite/internal/configuration/notifications/ConfigurationListenerTest.java

```diff
@@ -644,4 +667,366 @@ public void dataRace() throws Exception {
+
+        anyChild2.listen(configListener(ctx -> events.add("root.elements.any.elements2.any")));
+        anyChild2.i().listen(configListener(ctx -> events.add("root.elements.any.elements2.any.i")));
+
+        childCfg.elements2().any().listen(configListener(ctx -> events.add("root.elements.0.elements2.any")));
+        childCfg.elements2().any().i().listen(configListener(ctx -> events.add("root.elements.0.elements2.any.i")));
+
+        // Tests.
+        checkListeners(
+            () -> configuration.child().change(c -> c.changeStr("x").changeChild2(c0 -> c0.changeI(100))),
+            List.of("root", "root.child", "root.child.str", "root.child.child2", "root.child.child2.i"),
+            events
+        );
+
+        checkListeners(
+            () -> configuration.elements().get("0").str().update("x"),
+            List.of(
+                "root",
+                "root.elements",
+                "root.elements.onUpd",
+                //
+                "root.elements.any",
+                "root.elements.0",
+                //
+                "root.elements.any.str",
+                "root.elements.0.str"
+            ),
+            events
+        );
+
+        checkListeners(
+            () -> configuration.elements().get("0").elements2().get("0").i().update(200),
+            List.of(
+                "root",
+                "root.elements",
+                "root.elements.onUpd",
+                //
+                "root.elements.any",
+                "root.elements.0",
+                //
+                "root.elements.any.elements2",
+                "root.elements.0.elements2",
+                "root.elements.any.elements2.onUpd",
+                "root.elements.0.elements2.onUpd",
+                //
+                "root.elements.any.elements2.any",
+                "root.elements.0.elements2.any",
+                "root.elements.0.elements2.0",
+                //
+                "root.elements.any.elements2.any.i",
+                "root.elements.0.elements2.any.i",
+                "root.elements.0.elements2.0.i"
+            ),
+            events
+        );
+
+        checkListeners(
+            () -> configuration.elements().get("0").elements2().change(c -> c.create("1", doNothingConsumer())),
+            List.of(
+                "root",
+                "root.elements",
+                "root.elements.onUpd",
+                //
+                "root.elements.any",
+                "root.elements.0",
+                //
+                "root.elements.any.elements2",
+                "root.elements.0.elements2",
+                "root.elements.any.elements2.onCrt",
+                "root.elements.0.elements2.onCrt",
+                //
+                "root.elements.any.elements2.onUpd",
+                "root.elements.0.elements2.onUpd"
+            ),
+            events
+        );
+
+        checkListeners(
+            () -> configuration.elements()
```

Comment:
Should you call "any" listener in case of create operation? Seems logical, previous value is zero in this case. What do you think?

Reply:
I propose to discuss this in a separate ticket.

Reply:
I'll try to implement it.